src/test/test_verify.hpp

```
@@ -113,8 +110,57 @@ inline std::vector<prevail::RawProgram> read_elf_cached(const std::string& path,
 }
 
+// Compile-time test name truncation. Catch2's TEST_CASE accepts any const char*,
+// so we can pass .data from a constexpr char array instead of a string literal.
+// When the name exceeds MAX_LEN, we truncate and append a 4-digit hex hash of the
+// full name to preserve uniqueness.
+struct BoundedTestName {
+    static constexpr size_t MAX_LEN = 75;
+    char data[MAX_LEN + 1]{};
+
+    template <size_t N>
+    explicit constexpr BoundedTestName(const char (&str)[N]) {
```

Clang (14.0.6)

[warning] do not declare C-style arrays, use std::array<> instead

(modernize-avoid-c-arrays)